### DISCUSSION TITLE
Admin key should be hashed

### DISCUSSION THREAD

**Discussion by Mubelotix** [dani-garcia/vaultwarden#2887] [Ideas]
The admin key is passed to vaultwarden in plain text. This requires storing the key in plain text somewhere. This makes using vaultwarden as a systemd service unsafe. There could be an option to pass the hashed key

**Comment by BlackDex** [comment]
There are a few items on the list to improve upon regarding the admin login.
This could be one of them indeed.

Something like this has been on my mind for a while, yet, not got around to actually start adding more options to the admin login.
